Update documentation for Subscription v1alpha2 CRD (#16340)
kyma-project/kyma: docs/05-technical-reference/evnt-01-event-names.md

Event names depend on the type of event. Eventing supports the following event types:
- [CloudEvents](https://cloudevents.io/) - they use a specification for describing event data in a common way.
- legacy events - they are converted to CloudEvents by [Event Publisher Proxy](./00-architecture/evnt-01-architecture.md#event-publisher-proxy).

## Event name format

For a Subscription Custom Resource, the fully qualified event name takes the sample form of `order.created.v1` or `Account.Root.Created.v1`.

The event type is composed of the following components:
- Event: can have two or more segments separated by `.`; for example, `order.created` or `Account.Root.Created`
- Version: `v1`

For publishers, the event type takes this sample form:
- `order.created` or `Account.Root.Created` for legacy events coming from the `commerce` application
- `order.created.v1` or `Account.Root.Created.v1` for CloudEvents

## Event name cleanup

To conform to Cloud Event specifications, sometimes Eventing must modify the event name before dispatching an event.

### Special characters

> If the Event name contains any prohibited characters as per [NATS Jetstream specifications](https://docs.nats.io/running-a-nats-service/nats_admin/jetstream_admin/naming), the underlying Eventing services use a clean name with allowed characters only; for example, `system>prod*` becomes `systemprod`.

Reviewer: ```suggestion
If the Event's name contains any prohibited characters as per [NATS JetStream specifications](https://docs.nats.io/running-a-nats-service/nats_admin/jetstream_admin/naming), the underlying Eventing services use a clean name with allowed characters only; for example, `system>prod*` becomes `systemprod`.
```

Author: Updated.

> If the Event's name contains any prohibited characters as per [NATS JetStream specifications](https://docs.nats.io/running-a-nats-service/nats_admin/jetstream_admin/naming), the underlying Eventing services use a clean name with allowed characters only; for example, `system>prod*` becomes `systemprod`.

Reviewer 2: ```suggestion
If the event name contains any prohibited characters as per [NATS JetStream specifications](https://docs.nats.io/running-a-nats-service/nats_admin/jetstream_admin/naming), the underlying Eventing services use a clean name with allowed characters only; for example, `system>prod*` becomes `systemprod`.
```

> This could lead to a naming collision. For example, both `system>prod` and `systemprod` become `systemprod`. While this won't result in an error, it can cause Kyma to not work as expected. Take a look into this [troubleshooting guide](../04-operation-guides/troubleshooting/eventing/evnt-03-type-collision.md) for more information.

Reviewer: ```suggestion
This could lead to a naming collision. For example, both `system>prod` and `systemprod` become `systemprod`. While this won't result in an error, it can cause Eventing to not work as expected. Take a look into this [troubleshooting guide](../04-operation-guides/troubleshooting/eventing/evnt-03-type-collision.md) for more information.
```

what do you think ? I know it used to be this, but perhaps we could still edit it.

Author: Updated.

> This could lead to a naming collision. For example, both `system>prod` and `systemprod` become `systemprod`. While this won't result in an error, it can cause Eventing to not work as expected. Take a look into this [troubleshooting guide](../04-operation-guides/troubleshooting/eventing/evnt-03-type-collision.md) for more information.

Reviewer 2: ```suggestion
This can lead to a naming collision. For example, both `system>prod` and `systemprod` become `systemprod`. While this doesn't result in an error, it can cause Eventing to not work as expected. Take a look into this [troubleshooting guide](../04-operation-guides/troubleshooting/eventing/evnt-03-type-collision.md) for more information.
```
